dao2caom2/cleanup_augmentation.py

```diff
@@ -100,4 +100,4 @@ def visit(observation, **kwargs):
         f'Completed cleanup augmentation for {observation.observation_id}. '
         f'Remove {count} artifacts from the observation.'
     )
-    return {'artifacts': count}
+    return observation
```

Comment:
I'm not sure it's a good idea either, but these are the two reasons:
- I could not think of a better location to put the initial constructor call to SimpleObservation/DerivedObservation
- sometimes a SimpleObservation gets changed into a DerivedObservation, and then the two objects will not be the same. CFHT, GEMINICADC, GEMINI, have cases where there's no blueprint information to know it's a DerivedObservation, but there are other rules to check.

Having moved the initial construction out of `fits2caom2.py`, it's possible that the metadata that is required to decide on which type of Observation to construct is now readily available, so it may be possible to avoid the second case. If it's possible to avoid the second case, I will add a way to handle the first case too. I'm working on retrofitting this change to CFHT now, so I will see if I can work out a good pattern there, that can then be applied to all the other pipelines.